AsyncUsageAnalyzers/AsyncUsageAnalyzers.Test/Naming/UseAsyncSuffixUnitTests.cs

```diff
@@ -346,6 +346,32 @@ class ClassName
             await this.VerifyCSharpFixAsync(testCode, fixedCode, cancellationToken: CancellationToken.None).ConfigureAwait(false);
         }
 
+        [Fact]
+        public async Task TestReturnGenericValueTaskAsync()
+        {
+            string testCode = @"
+using System.Threading.Tasks;
+class ClassName
+{
+    ValueTask<int> FirstMethod() { return new ValueTask<int>(3); }
+    ValueTask<int> SecondMethodAsync() { return new ValueTask<int>(3); }
```

Comment:
How about a test for the non-generic `ValueTask` type as well?

Reply:
It's not available in the **System.Threading.Tasks.Extensions** package and requires the test to compile against .NET Core. The test project doesn't currently support this to my knowledge.

Reply:
The non-generic `ValueTask` struct certainly *is* available from that same package, and it does not require .NET Core as a target. 
In my Nerdbank.Streams project I use `ValueTask` in many places. Hitting F12 on it reveals it is defined here:
`.nuget\packages\system.threading.tasks.extensions\4.5.0\ref\netstandard1.0\System.Threading.Tasks.Extensions.dll`

Reply:
👍 I was going by these pages:

https://apisof.net/catalog/System.Threading.Tasks.ValueTask%3CTResult%3E
https://apisof.net/catalog/System.Threading.Tasks.ValueTask

Looks like it hasn't been updated for the 4.5.0 release of these packages.

Reply:
Ya, apisof.net has been the most accurate reference for this info (much better than msdn/docs). But it does appear to be outdated now. :(